### ISSUE TITLE
queryParameters from nested traits is not honored in generated code

### ISSUE THREAD

**Issue by techpavan** [phoenixnap/springmvc-raml-plugin#287]
Found that applying the workaround in #283 by using `required: false` rather than short-hand notation is effective only when traits are directly referenced. If a trait is referring to a different trait which has queryParameters, the nested trait's queryParameters do not reflect in the generated code.

Reproducer - Clone the code from https://github.com/techpavan/springmvc-raml-plugin-issues/ and check the code generated to list all the bus brands against for the car brands.

**Comment by stojsavljevic** [comment]
Can you please check if the issue is related to `raml-parser-2`?
There is a link to `raml-parser-2` bug in #283 where you can find the way to reproduce issues with the parser.

**Comment by techpavan** [comment]
Right. Its an issue from the `raml-parser-2` project. Have submitted an issue there.

https://github.com/raml-org/raml-java-parser/issues/558
